src/Control/Lens/Cons.hs

```diff
@@ -319,6 +323,42 @@ _tail :: Cons s s a a => Traversal' s s
 _tail = _Cons._2
 {-# INLINE _tail #-}
 
+-- | Convert one type with a 'Cons' instance into the other.
+--
+-- Rewrite rules are provided for efficient conversion between
+-- 'String' and 'Text', @['Word8']@ and 'ByteString', and lazy and
+-- strict 'Text' and 'ByteString'.  Programs must be compiled
+-- with @-O@ to use them.
+--
+-- If @(recons . recons)@ exists it should obey:
+--
+-- > recons . recons ≡ id
```

Comment:
@frasertweedale `recons = id` if `s1 = s2`, isn't it? Maybe that can be written explcitly (And instances added).